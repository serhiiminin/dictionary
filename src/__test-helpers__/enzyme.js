import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
