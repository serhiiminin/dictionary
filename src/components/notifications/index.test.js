import React from "react";
import { mount } from 'enzyme';
import muiTheme from "../../root/mui-theme";
import Notifications from "./component";

describe("Notifications components", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Notifications notifications={[]} hideNotification={() => {}}>
        children
      </Notifications>,
      muiTheme
    );
  });
  test("render empty", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("render with one item simulate and to hide", () => {
    const hideNotification = jest.fn();
    wrapper.setProps({
      notifications: [
        {
          id: "id",
          text: "text",
          type: "success"
        }
      ],
      hideNotification
    });
    wrapper.find("button").simulate("click");
    expect(hideNotification).toHaveBeenCalled();
  });
});
