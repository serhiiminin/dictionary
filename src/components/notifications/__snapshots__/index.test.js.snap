// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications components render empty 1`] = `
<Notifications
  classes={Object {}}
  hideNotification={[Function]}
  notifications={Array []}
>
  children
  <ul />
</Notifications>
`;
